File: worklist-ui/src/main/java/com/gcgenome/lims/client/usecase/work/UpdateSerialEvent.java
package com.gcgenome.lims.client.usecase.work;

import com.gcgenome.lims.client.domain.PanelEvent;
import com.gcgenome.lims.client.domain.Serial;

import java.util.function.Consumer;

public interface UpdateSerialEvent {
    void subscribe(Consumer<PanelEvent<Serial>> consumer);
}

File: domain/src/main/kotlin/com/gcgenome/lims/domain/event/UpdateSerialEvent.kt
package com.gcgenome.lims.domain.event

import com.gcgenome.lims.domain.Serial
import java.util.*

data class UpdateSerialEvent (
    private val id: UUID,
    private val param: Serial,
    private val type: PanelEvent.Type = PanelEvent.Type.UPDATE_SERIAL
): PanelEvent<Serial, UpdateSerialEvent> {
    override fun id(): UUID = id
    override fun type(): PanelEvent.Type = type
    override fun param(): Serial = param
}